x/operator/keeper/genesis.go

```diff
@@ -9,14 +10,96 @@ import (
 func (k Keeper) InitGenesis(ctx sdk.Context, state types.GenesisState) []abci.ValidatorUpdate {
 	for i := range state.Operators {
 		op := state.Operators[i] // avoid implicit memory aliasing
-		if err := k.SetOperatorInfo(ctx, op.EarningsAddr, &op); err != nil {
-			panic(err)
+		if op.OperatorInfo.EarningsAddr == "" {
+			op.OperatorInfo.EarningsAddr = op.OperatorAddress
 		}
+		if err := k.SetOperatorInfo(ctx, op.OperatorAddress, &op.OperatorInfo); err != nil {
+			panic(errorsmod.Wrap(err, "failed to set operator info"))
+		}
+	}
+	for _, record := range state.OperatorRecords {
+		addr := record.OperatorAddress
+		// #nosec G703 // already validated
+		operatorAddr, _ := sdk.AccAddressFromBech32(addr)
+		for _, detail := range record.Chains {
+			wrappedKey := types.NewWrappedConsKeyFromHex(detail.ConsensusKey)
+			bz := k.cdc.MustMarshal(wrappedKey.ToTmProtoKey())
+			k.setOperatorConsKeyForChainIDUnchecked(ctx, operatorAddr, wrappedKey.ToConsAddr(), detail.ChainID, bz)
+		}
+	}
+	// init the state from the general exporting genesis file
+	err := k.SetAllOptedInfo(ctx, state.OptStates)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all opted info"))
+	}
+	err = k.SetAllOperatorUSDValues(ctx, state.OperatorUSDValues)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all operator USD values"))
+	}
+	err = k.SetAllAVSUSDValues(ctx, state.AVSUSDValues)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all AVS USD values"))
+	}
+	err = k.SetAllSlashStates(ctx, state.SlashStates)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all slash info"))
+	}
+	err = k.SetAllPrevConsKeys(ctx, state.PreConsKeys)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all previous consensus keys"))
+	}
+	err = k.SetAllOperatorKeyRemovals(ctx, state.OperatorKeyRemovals)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all key removals for operators"))
 	}
 	return []abci.ValidatorUpdate{}
 }
 
-func (Keeper) ExportGenesis(sdk.Context) *types.GenesisState {
-	// TODO
-	return types.DefaultGenesis()
+func (k Keeper) ExportGenesis(ctx sdk.Context) *types.GenesisState {
```

Comment:
just record error, should we  assign data when it is successful without any error ?

Reply:
I am unsure if an error should influence the export of other fields. I can adjust it to return a null if there is any error.

Reply:
Use panic to address the error according to Max's advice.